flume-ng-sources/flume-kafka-source/src/main/java/org/apache/flume/source/kafka/KafkaSource.java

```diff
@@ -464,12 +468,17 @@ private String lookupBootstrap(String zookeeperConnect, SecurityProtocol securit
     ZkUtils zkUtils = ZkUtils.apply(zookeeperConnect, ZK_SESSION_TIMEOUT, ZK_CONNECTION_TIMEOUT,
         JaasUtils.isZkSecurityEnabled());
     try {
-      List<BrokerEndPoint> endPoints =
-          asJavaListConverter(zkUtils.getAllBrokerEndPointsForChannel(securityProtocol)).asJava();
+      List<Broker> brokers = seqAsJavaList(zkUtils.getAllBrokersInCluster());
+
       List<String> connections = new ArrayList<>();
-      for (BrokerEndPoint endPoint : endPoints) {
-        connections.add(endPoint.connectionString());
+
+      for (Broker broker : brokers) {
+        List<EndPoint> endPointList = seqAsJavaList(broker.endPoints());
+        for (EndPoint endPoint : endPointList) {
+          connections.add(endPoint.host() + ":" + endPoint.port());
```

Comment:
I'm intrigued by this change - is endPoint.connectionString no longer supplied? It seems the old version would have been better.

Reply:
`endPoint.connectingString` does exist in the new library, but the newer one returns additional text.  This is evident from the test `TestKafkaSource#testBootstrapLookup`.  The test fails with the following output.  

Expected :localhost:65069
Actual   :PLAINTEXT://localhost:65069

Reply:
Is there something we need to be aware of here when having non-plaintext listeners?

Reply:
Sorry @tmgstevens I have no idea about that.  The test fails with this error if I change it to `endPoint.connectingString`

[ERROR] Tests run: 36, Failures: 11, Errors: 0, Skipped: 0, Time elapsed: 311.043 s <<< FAILURE! - in org.apache.flume.source.kafka.TestKafkaSource
[ERROR] testBootstrapLookup(org.apache.flume.source.kafka.TestKafkaSource)  Time elapsed: 4.101 s  <<< FAILURE!
junit.framework.ComparisonFailure: expected:<[]localhost:51213> but was:<[PLAINTEXT://]localhost:51213>
	at org.apache.flume.source.kafka.TestKafkaSource.testBootstrapLookup(TestKafkaSource.java:607)

Reply:
I think this looks okay, but I'd like to test it in a secure environment.

Reply:
On reflection I'm not happy with this. I haven't had the chance to test it in a secure environment, but I think the test is wrong. It is going to be important to prefix with the type of listener that is being used. For the test we can prefix PLAINTEXT, but if it's SASL_SSL, or any other type of listener then I think we'd need to provide that back from there.

Reply:
@tmgstevens  apologies for the delay. I am happy to change the test in that case, but I have no idea about testing other types / neither do I have the environment to test the rest of the types.  Any advise?

Reply:
Please could you try this code:
```
private String lookupBootstrap(String zookeeperConnect, SecurityProtocol securityProtocol) {
    ZkUtils zkUtils = ZkUtils.apply(zookeeperConnect, ZK_SESSION_TIMEOUT, ZK_CONNECTION_TIMEOUT,
        JaasUtils.isZkSecurityEnabled());
    try {
      Seq<Broker> allBrokersInCluster = zkUtils.getAllBrokersInCluster();
      List<Broker> brokerList = JavaConverters.seqAsJavaListConverter(zkUtils.getAllBrokersInCluster()).asJava();
      List<BrokerEndPoint> endPoints = brokerList.stream()
              .map(broker -> broker.getBrokerEndPoint(
                  ListenerName.forSecurityProtocol(securityProtocol))
              )
              .collect(Collectors.toList());
      List<String> connections = new ArrayList<>();
      for (BrokerEndPoint endPoint : endPoints) {
        connections.add(endPoint.connectionString());
      }
      return StringUtils.join(connections, ',');
    } finally {
      zkUtils.close();
    }
  }
```